File: src/main/java/com/wanderers/hotelier_webservices/server/exception/BookingUnavailableException.java
package com.wanderers.hotelier_webservices.server.exception;

/**
 * Custom Exception class which used to throw when bookings are not available
 */
public class BookingUnavailableException extends Exception {

    private static final long serialVersionUID = 1L;

    /**
     * Class constructor
     *
     * @param message String value of Exception
     */
    public BookingUnavailableException(String message) {
        super(message);
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/exception/BookingUnavailableException.kt
package com.wanderers.hotelier_webservices.server.exception

/**
 * Custom Exception class which used to throw when bookings are not available
 */
class BookingUnavailableException
/**
 * Class constructor
 *
 * @param message String value of Exception
 */
    (message: String?) : ServerException(message) {
    companion object {
        private const val serialVersionUID = 1L
    }
}